src/sync/sync_session.hpp

```diff
@@ -263,6 +266,17 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     std::string m_realm_path;
     _impl::SyncClient& m_client;
 
+    enum class CompletionWaitType {
+        Upload, Download, Sync,
+    };
+
+    // For storing wait-for-completion requests if the session isn't yet ready to handle them.
+    struct CompletionWaitPackage {
+        CompletionWaitType type;
```

Comment:
Since the only thing we use this enum for is to determine which function to call, we could instead directly store a member function pointer and avoid having to switch on the enum.